backend/src/main/kotlin/xyz/elitese/ehrenamtskarte/database/repos/AcceptingStoresRepository.kt

```diff
@@ -19,26 +17,38 @@ object AcceptingStoresRepository {
     // TODO Fulltext search is possible with tsvector in postgres: https://www.compose.com/articles/mastering-postgresql-tools-full-text-search-and-phrase-search/
     // TODO Probably not relevant right now
     fun findBySearch(searchText: String?, categoryIds: List<Int>?): SizedIterable<AcceptingStoreEntity> {
+        val categoryMatcher =
+            if (categoryIds == null)
+                Op.TRUE
+            else
+                Op.build { AcceptingStores.categoryId inList categoryIds }
+
+        val textMatcher =
+            if (searchText == null)
+                Op.TRUE
+            else
+                OrOp(
+                    listOf(
+                        AcceptingStores.name ilike "%$searchText%",
+                        AcceptingStores.description ilike "%$searchText%",
+                        Addresses.location ilike "%$searchText%",
+                        Addresses.postalCode ilike "%$searchText%",
+                        Addresses.street ilike "%$searchText%"
+                    )
+                )
 
-        val matchCategoryIds = if (categoryIds != null) Op.build { AcceptingStores.categoryId inList categoryIds } else Op.TRUE
-        val matchSearchText = if (searchText != null) {
-            val lowerCaseSearchText = searchText.toLowerCase()
-            Op.build {
-                ((AcceptingStores.name.lowerCase() like "%${lowerCaseSearchText}%") or
-                        (AcceptingStores.description.lowerCase() like "%${lowerCaseSearchText}%") or
-                        exists(PhysicalStores.select(
-                            PhysicalStores.storeId eq AcceptingStores.id and
-                                    exists(Addresses.select(
-                                        Addresses.id eq PhysicalStores.addressId and
-                                                (Addresses.location.lowerCase() like "%${lowerCaseSearchText}%") or
-                                                (Addresses.postalCode.lowerCase() like "%${lowerCaseSearchText}%") or
-                                                (Addresses.street.lowerCase() like "%${lowerCaseSearchText}%")
-                                    ))
-                        )))
-            }
-        } else Op.TRUE
-        return AcceptingStoreEntity.find {
-            matchSearchText.and(matchCategoryIds)
-        }
+        return (AcceptingStores leftJoin PhysicalStores leftJoin Addresses)
+            .slice(AcceptingStores.columns)
```

Comment:
Slice is like a SQL SELECT? If we throw away the Columns of Address, does `Addresses.location ilike "%$searchText%"` etc. still work?

Reply:
As you can see in the SQL Query in the PR description, the answer is yes :)